indico/web/menu.py

```diff
@@ -84,3 +86,94 @@ def __repr__(self):
     @property
     def active(self):
         return request.endpoint in self.endpoints
+
+
+class SideMenuSection(object):
+    """Defines a side menu section (item set).
+
+    :param title: the title of the section (displayed)
+    :param weight: the "weight" (higher means it shows up first)
+    :param active: whether the section should be shown expanded by default
+    :param visible: whether the section should be shown at all
+    :param event_feature: associated feature (id), section will be hidden
+                          if feature is disabled
+    :param icon: icon that will be displayed next to the section title.
+    """
+
+    def __init__(self, title, weight=-1, active=False, visible=True, event_feature=None, icon=None):
+        self.title = title
+        self._active = active
+        self.visible = visible
+        self.event_feature = event_feature
+        self._items = set()
+        self.icon = 'icon-' + icon
+        self.weight = weight
+
+    def add_item(self, item):
+        self._items.add(item)
+
+    @property
+    def items(self):
+        return sorted(self._items, key=lambda x: (-x.weight, x.title))
+
+    @property
+    def active(self):
+        return self._active or any(item.active for item in self._items)
+
+
+class SideMenuItem(object):
+    """Defines a side menu item.
+
+    :param title: the title of the menu item (displayed)
+    :param url: the URL that the link will point to
+    :param weight: the "weight" (higher means it shows up first)
+    :param active: whether the item will be shown as active by default
+    :param visible: whether the item should be shown at all
+    :param event_feature: associated feature (id), the item will be hidden
+                          if feature is disabled
+    :param section: section the item will be put in
+    :param icon: icon that will be displayed next to the item
+    """
+
+    def __init__(self, title, url, weight=-1, active=False, disabled=False, visible=True, event_feature=None,
+                 section=None, icon=None):
+        self.title = title
+        self.url = url
+        self.active = active
+        self.disabled = disabled
+        self.visible = visible
+        self.event_feature = event_feature
+        self.section = section
+        self.weight = weight
+        self.icon = ('icon-' + icon) if icon else None
+
+
+def menu_from_signals(menu_id, active_item=None, **kwargs):
```

Comment:
is the name still a good fit? (it builds it from signals, but that's more of an implementation detail)